containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -465,26 +476,24 @@ class UnorderedMap {
     return is_insertable_map ? get_flag(erasable_idx) : false;
   }
 
-  bool begin_erase() {
+  bool begin_erase(const execution_space &space = execution_space{}) {
     bool result = !erasable();
     if (is_insertable_map && result) {
-      execution_space().fence(
+      space.fence(
           "Kokkos::UnorderedMap::begin_erase: fence before setting erasable "
           "flag");
       set_flag(erasable_idx);
     }
     return result;
   }
 
-  bool end_erase() {
+  bool end_erase(const execution_space &space = execution_space{}) {
     bool result = erasable();
     if (is_insertable_map && result) {
-      execution_space().fence(
-          "Kokkos::UnorderedMap::end_erase: fence before erasing");
-      Impl::UnorderedMapErase<declared_map_type> f(*this);
-      f.apply();
-      execution_space().fence(
-          "Kokkos::UnorderedMap::end_erase: fence after erasing");
+      space.fence("Kokkos::UnorderedMap::end_erase: fence before erasing");
```

Comment:
Same here.

Reply:
Ok, I'll remove the fences. Should I provide an overload that does not expect an execution space instance?